pyproject.toml

```diff
@@ -47,12 +47,10 @@ optional = true
 [tool.poetry.group.testing]
 optional = true
 
-[tool.poetry.group.linting]
-optional = true
-
 [tool.poetry.group.development.dependencies]
 pyspark = ">3"
 semver = "^3"
+pre-commit = "^3.7.1"
```

Comment:
As I may understand we are using the latest version in CI, but the pinned one in virtualenv. You explained well, why it is better to use pip+latest version in CI, let's use "*" instead of the version here in this case.